### ISSUE TITLE
Fetch OpenAPI Spec from remote

### ISSUE THREAD

**Issue by danielkocot** [openapi-processor/openapi-processor-gradle#21]
Hi,

by the way great effort so far. How can I make the 'processSpring' depend to another task that has to be done first? Because I would like to download the spec within build step. 

Best,
Daniel

**Comment by hauner** [comment]
Thanks :-)

in theory you should be able to use an http(s) url as `apiPath`. This worked for me on a simple remote api with `$ref`s. This will probably not work if you need authorization.


I played a bit with gradle and creating a task dependency works like this:

ˋˋˋgroovy
tasks.register('foo') {
    doLast {
        println 'foo'
    }
}

afterEvaluate {
    tasks.processSpring.dependsOn('foo')
}
ˋˋˋ

The plugin creates the `processSpring` task in an `afterEvalute` block. It looks like it is not visible outside/before `afterEvaluate`.

With all the lazy task creation I did not expect this. ;-)

**Comment by danielkocot** [comment]
If I set a url as value of `apiPath` I get an error message `Cannot convert URL to a file`. I will have a look at this task dependency thing. Thanks!
